### PR TITLE
[#8] Conditional compilation: disable-ekg flag.

### PR DESCRIPTION
**Pull request by denisshevchenko**
description
-----------

- [x] flag `disable-ekg` controls whether the EKG subsystem is included in compilation

checklist
---------

- [x] compiles (`cabal new-clean; cabal new-build`)
- [x] tests run successfully (`cabal new-test`)
- [x] documentation added and created (`cd docs; nix-shell --run make`)
- [x] link to an issue
- [x] link to an epic
- [x] add estimate points
- [x] add milestone (the same as the linked issue)

### CODE REVIEW

File: test/Cardano/BM/Test/Configuration.lhs [hunk 1]
```diff
@@ -249,7 +258,7 @@ unitConfigurationParsedRepresentation = do
             , "    - monitor: ((time > (23 s)) Or (time < (17 s)))"
             , "    - actions:"
             , "      - AlterMinSeverity \"chain.creation\" Debug"
-            , "    '#aggregation.critproc.observable':"
+            , "    ! '#aggregation.critproc.observable':"
```
Comment (CodiePP) [inline test/Cardano/BM/Test/Configuration.lhs]: this probably has to do with an outdated libyaml. let's find out..

Comment (denisshevchenko) [reply test/Cardano/BM/Test/Configuration.lhs]: Without this `!` id soesn't work.. :-(

File: src/Cardano/BM/Data/Configuration.lhs [hunk 2]
```diff
@@ -88,6 +90,10 @@ implicit_fill_representation =
         case hasEKG r of
         Nothing -> r
         Just _  -> r {setupBackends = setupBackends r <> [EKGViewBK]}
+#else
+    remove_ekgview_if_not_defined r = r
+    add_ekgview_if_port_defined r = r
```
Comment (CodiePP) [inline src/Cardano/BM/Data/Configuration.lhs]: or `add_ekgview_if_port_defined = id` ?

Comment (denisshevchenko) [reply src/Cardano/BM/Data/Configuration.lhs]: Yep. :-) Should I fix?

File: test/Cardano/BM/Test/Configuration.lhs (hunk 1 above)
Comment (CodiePP) [reply test/Cardano/BM/Test/Configuration.lhs]: right, we have to recreate consistency among compilation with nix, cabal and stack.

File: src/Cardano/BM/Data/Configuration.lhs (hunk 2 above)
Comment (denisshevchenko) [reply src/Cardano/BM/Data/Configuration.lhs]: Done.

**Pull request merged**